libethereum/Client.cpp

```diff
@@ -103,15 +103,22 @@ void Client::init(p2p::Host* _extNet, std::string const& _dbPath, WithExisting _
 	m_bq.setOnBad([=](Exception& ex){ this->onBadBlock(ex); });
 	bc().setOnBad([=](Exception& ex){ this->onBadBlock(ex); });
 
+
 	if (_forceAction == WithExisting::Rescue)
 		bc().rescue(m_stateDB);
 
 	m_gp->update(bc());
 
 	auto host = _extNet->registerCapability(make_shared<EthereumHost>(bc(), m_stateDB, m_tq, m_bq, _networkId));
 	m_host = host;
+
+	bc().setOnBlockImport([=](BlockHeader const& _info){
```

Comment:
Now you can move it above to line 105, because m_host exists there, and this callback can't be called, I think, until `startWorking()` call in the end of this method, and anyway if condition in lambda won't be true until m_host is set.